server/api/api_accessor.py

```diff
@@ -28,47 +28,45 @@ def __init__(self):
     def update_token(self, token):
         self.token = token
 
-    def __enter__(self):
-        if not self.session:
-            try:
-                self.session = OAuth2Session(
-                    client=self.client,
-                    token=self.token,
-                    auto_refresh_url=API_TOKEN_URI,
-                    auto_refresh_kwargs={
-                        "client_id": API_CLIENT_ID,
-                        "client_secret": API_CLIENT_SECRET
-                    },
-                    token_updater=self.update_token
-                )
-                self.token = self.session.fetch_token(
-                    token_url=API_TOKEN_URI,
-                    client_id=API_CLIENT_ID,
-                    client_secret=API_CLIENT_SECRET
-                )
-            except:
-                self.__exit__(*sys.exc_info())
-
-        return self.session
-
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        if exc_type == MissingTokenError:
+    def get_session(self):
+        if self.session is not None:
+            return self.session
+        try:
+            session = OAuth2Session(
+                client=self.client,
+                token=self.token,
+                auto_refresh_url=API_TOKEN_URI,
+                auto_refresh_kwargs={
+                    "client_id": API_CLIENT_ID,
+                    "client_secret": API_CLIENT_SECRET
+                },
+                token_updater=self.update_token
+            )
+            token = self.session.fetch_token(
+                token_url=API_TOKEN_URI,
+                client_id=API_CLIENT_ID,
+                client_secret=API_CLIENT_SECRET
+            )
+        except MissingTokenError:
             self.logger.error("There was an error while connecting the API - token is missing or invalid")
-
-        elif exc_type == InsecureTransportError:
+            raise ConnectionError
+        except InsecureTransportError:
             self.logger.error(
                 "API (%s,%s) should be HTTPS, not HTTP. Enable OAUTHLIB_INSECURE_TRANSPORT to avoid this warning.",
                 config.API_BASE_URL,
                 config.API_TOKEN_URI
             )
-
-        elif exc_type == SSLError:
+            raise ConnectionError
+        except SSLError:
             self.logger.error("The certificate verification failed while connecting the API")
+            raise ConnectionError
+        except Exception as e:   # TODO do we need this?
```

Comment:
I think the answer to this question is 'no'

Reply:
@Rackover disagreed on that one.

Reply:
ATM I think it's better if we log it as a general exception and eventually if we see a specific exception occurs frequently as part of normal functioning we can add the except case

Reply:
Well then remove the todo comment

Reply:
Sounds to me like you have a good reason for what you’re doing so there’s no need to add a comment bringing it into question